.github/workflows/json_librarian.yaml

```diff
@@ -0,0 +1,30 @@
+name: JSON Library Sorter and Doc Gen
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - "main"
+    paths:
+      - "library/library.json"
+      - ".github/workflows/json_librarian.yaml"
+      - ".github/workflows/scripts/library_doc/**"
+
+jobs:
+  librarian:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/astral-sh/uv:trixie-slim
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v5
+
+      - name: Update Library Files
+        run: |
+          uv run ${{ github.workspace }}/.github/scripts/library_doc/update_library_files.py
```

Comment:
There seems to be a path issue when running this.
I have a test repo for these actions and I've copied your changes into that.
They're tricky to get right, I've just invited you as a collaborator to that repo so you can test easily there.

Output of error
https://github.com/andrew-codechimp/bn-device-test/actions/runs/18994661549/job/54252756574

Reply:
resolved after a bunch of testing! basically: created a library and docs dependency group to be used exclusively in the workflow runs where theyre needed. that way those dependencys arent being added to the default dev environment, since theyre only ever used in those workflows. this also has the benefit of not installing homeassistant in the workflow (this was the case before the way you had the workflows set up and I introduced that oddity when I added everything to the dev group)

the dependency issue you were seeing had nothing to do with any of that however, and instead was due to my attempt at using a UV container.. which had a lot of unintended consequences. using a default image with the setup-uv action is simpler all around so I switched back to that.